modules/tide_content_collection/tide_content_collection.module

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @file
+ * Contains tide_content_collection.module.
+ */
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\paragraphs\ParagraphInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function tide_content_collection_paragraph_access(ParagraphInterface $entity, $operation, AccountInterface $account) {
+  $permissions = &drupal_static(__FUNCTION__, []);
```

Comment:
@joshua-salsadigital why do you need static cache here?
The logic of this hook is very straightforward:
* if the bundle is `content_collection` and the operator is `[update, delete]`:
  + allowed if `access content_collection paragraph`
* otherwise return`neutral`
* DO NOT MESS WITH THE PERMISSION SYSTEM with your own static cache

Reply:
Corrected the logic, if valid type and access for permission check only.